Add error handler for transactional event

hyh1016 opened the pull request:
Exceptions propagated from the transactional event listener are handled by `TransactionSynchronizationUtils` and are not rethrown.
However, the lack of a configurable error handler for these exceptions makes it difficult to customize exception handling during transactional event processing.
To address this, I added an abstract error handler that can be defined as follows:

```java
@Component
public abstract class TransactionalEventErrorHandlerImpl extends TransactionalEventErrorHandler {

	// can inject other beans

	@Override
	public void handle(ApplicationEvent event, @Nullable Throwable ex) {
		// custom logic
	}

}
```

hyh1016 committed:
Add global error handler for transactional event listener

Signed-off-by: YIHYUN HA <hihyun16@gmail.com>

pivotal-cla commented:
@hyh1016 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=34146)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=34146) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@hyh1016 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=34146)!

snicoll commented:
@hyh1016 thanks for the suggestion but we can't consider this PR until you add some unit tests that demonstrates what this is meant to fix. Can you please amend your proposal with them?

hyh1016 commented:
@snicoll Okay, thanks for the feedback. I added unit tests to demonstrate how the code I added works.
Also, while writing the unit tests, I found an issue in the existing logic, so I updated the original PR and pushed the changes again.
